### DISCUSSION TITLE
Nodes overlapping each other

### DISCUSSION THREAD

**Discussion by jenish025** [xyflow/xyflow#2293] [Q&A] [answered]
Quick question regarding node overlapping.
Is there a way to make the nodes not overlap? (see diagram below)
![nodeoverlapping](https://user-images.githubusercontent.com/109336900/179193645-523db295-f150-426e-ba93-11c48ca2bd78.PNG)

**Comment by Ipicky22** [comment, marked as answer]
Hi @jenish025,
You can implement a dagre tree on your graph, it doesn't solve the problem of overlap, but with a button you can have a better display for your graph without overlap.
[Documentation v10 dagre treet](https://reactflow.dev/docs/examples/layout/dagre/)

**Reply by moklick** [reply]
Thanks @Ipicky22 ! It's not built in, but you can use any layouting library that fit your needs:

* elkjs https://github.com/kieler/elkjs
* d3-hierarchy https://github.com/d3/d3-hierarchy
* dagre https://github.com/dagrejs/dagre
